lib/concurrent/collection/array.rb

```diff
@@ -0,0 +1,38 @@
+require 'concurrent/utility/engine'
+require 'concurrent/thread_safe/util'
+
+module Concurrent
+  if Concurrent.on_cruby?
+
+    # Because MRI never runs code in parallel, the existing
+    # non-thread-safe structures should usually work fine.
+
+    # @!macro [attach] concurrent_array
+    #
+    #   A thread-safe subclass of Array. This version locks against the object
+    #   itself for every method call, ensuring only one thread can be reading
+    #   or writing at a time. This includes iteration methods like `#each`.
+    #
+    #   @see http://ruby-doc.org/core-2.2.0/Array.html Ruby standard library `Array`
+    class Array < ::Array;
```

Comment:
In the 0.9 release I reconfigured a bunch of classes/modules to conform to more idiomatic namespacing. This change seems like a step back. I have no issues moving files from the `thread_safe` directory into the `collection` directory, but why move the top-level classes into there as well? Is there something I'm missing?

Unless there is a compelling reason to do otherwise, I'd like to keep array.rb, hash.rb, tuple.rb, and map.rb at the top level. We should be able to move all of the backend code from map.rb into a new file w/i the collection folder and namespace those classes idiomatically.